build/buildPackage.sh

```diff
@@ -127,6 +126,15 @@ manifest="
 
 "
 
+if [ "$#" -lt "1" ]; then
```

Comment:
> Right I'll check for renderman artefacts and only package renderman if it's required? Also should we do the same for Arnold?

Just to be doubly clear, the purpose of this repository is to provide a means of packaging _all_ of Gaffer's dependencies, for people developing outside of Image Engine (or any other custom build environment that takes care of things). It's also a record of how each public release of Gaffer itself was built. The packages built by `buildPackage.sh` are for distributing the resulting binaries to people who don't want to build them, and the `build*.sh` scripts are to help build those binaries, or for people who want finer control over their own builds.

I can't see any reason we'd want to distribute an incomplete package, so the option of creating incomplete packages is tangential to the goal. So, if this "without renderman" feature is useful to you and defaults off, then that's OK, but there's no need to add a "without arnold" option for completeness. The whole point is to make a single, complete, known configuration for everyone to work from...

> How do you feel replacing each of the shell scripts with a python script?

I feel like that would be a less readable way of running a bunch of shell commands, and that a python based system would have to provide some additional higher level benefits to justify itself...